src/loaders/compute_tools/tool_result_parser.py

```diff
@@ -0,0 +1,105 @@
+import copy
+import os
+from pathlib import Path
+from typing import Dict, List, Set
+
+import jsonlines
+import pandas as pd
+
+from src.loaders.compute_tools.tool_common import GenomeTuple
+
+# used as collection_and_field_names.FLD_KBASE_ID
+TOOL_KBASE_ID = "kbase_id"
+TOOL_GENOME_ATTRI_FILE = "genome_attribs.jsonl"
+
+
+def create_jsonl_files(
+        file_path: Path,
+        docs: list) -> None:
+    """
+    create jsonl file from the list of documents
+
+    :param file_path: the path to the jsonl file
+    :param docs: the list of documents
+    """
+    print(f'Creating JSONLines import file: {file_path}')
+    with jsonlines.open(file_path, mode='w') as writer:
+        writer.write_all(docs)
+
+
+def read_genome_attri_result(
+        batch_result_dir: Path,
+        tool_file_name: str,
+        features: Set[str],
+        genome_id_col: str,
+        ids_to_files: Dict[str, GenomeTuple],
+        prefix: str = '') -> List[Dict[str, str]]:
+    """
+    process the output file generated by the tool (checkm2, gtdb-tk, etc) to create a format suitable for
+    importing into ArangoDB
+
+    :param batch_result_dir: the directory where the tool result files are stored
+    :param tool_file_name: the name of the tool result file
+    :param features: a list of features to be retrieved from the tool result file
+    :param genome_id_col: the column name of the genome id in the tool result file
+    :param ids_to_files: a dictionary of tool genome ids to the corresponding GenomeTuple
+    :param prefix: the prefix to append to the genome id from the tool result file
+
+    :return: a dictionary of genome attributes
+    """
+    tool_genome_map = {tool_id: genome_tuple.data_id for tool_id, genome_tuple in ids_to_files.items()}
+
+    tool_file = os.path.join(batch_result_dir, tool_file_name)
+    docs = list()
+    if os.path.exists(tool_file):
```

Comment:
I'm sure we discussed this before but silently ignoring missing files makes me nervous. What was the reasoning behind this again?

Reply:
raise error if file doesn't exist. I copy the logic from parse_tool_result.py. I don't remember exactly, but it's probably due to gtdb_tk can produce both or either result files. I think I will just make sure when it's used by gtdb_tk, I will check the result file existence.

Reply:
Maybe add a bool flag to determine whether to throw an error if the file is missing or not. Then it's clear from the signature what's going to happen

Reply:
👍

Reply:
:+1: